cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -663,7 +665,30 @@ func newStateMachine(disk disks.Disk) *stateMachine {
 func (m *stateMachine) execute() (finished bool, err error) {
 	next, err := m.current()
 	m.current = next
-	return next == nil, err
+	finished = next == nil
+	if finished && err == nil {
+		if err := m.finalize(); err != nil {
+			return true, err
+		}
+	}
+	return finished, err
+}
+
+func (m *stateMachine) finalize() error {
```

Comment:
@anonymouse64 I don't feel particularly strongly either way, it might make sense if you the partital copying ?

on a note about stateMachine,  I didn't think of it when I reviewed the code, but maybe it should be called "recoverMode(State)Machine) ?

Reply:
yeah the name of the struct seems awfully generic but I didn't want to spend time bike-shedding on the name, I think `recoverModeStateMachine` is fine if a bit verbose